src/auth.ts

```diff
@@ -25,4 +28,28 @@ async function signOut({ returnTo }: { returnTo?: string } = {}) {
   await terminateSession({ returnTo });
 }
 
-export { getSignInUrl, getSignUpUrl, signOut };
+export async function switchToOrganization(
+  organizationId: string,
+  options: SwitchToOrganizationOptions = {},
+): Promise<UserInfo> {
+  const { returnTo, revalidationStrategy = 'path', revalidationTags = [] } = options;
+  const headersList = await headers();
+  const pathname = returnTo || headersList.get('x-url') || '/';
+  const result = await refreshSession({ organizationId, ensureSignedIn: true });
+
```

Comment:
I think this chunk is something we would want to handle here
```ts
      const args = {
        organizationId,
        clientId: process.env.WORKOS_CLIENT_ID!, // probably have this within this package already
        provider: "authkit", // ?
        redirectUri: pathname,
      };

      if (err.error === "sso_required" || err.error === "mfa_enrollment") {
        const url = workos.userManagement.getAuthorizationUrl({
          ...args,
        });
        redirect(url);
      }
```

It ensures that we send them to the correct re-auth when switching fails. cc: @alisherry who has the right intersection of knowledge to discuss more clearly.

Reply:
Looking at refreshSession would it ever throw an error with `err.error` equaling one of the above?

https://github.com/workos/authkit-nextjs/blob/c6de3414418c486f561264668ad0f2fc53826953/src/session.ts#L254-L325

Reply:
@alisherry Thanks, I updated the PR with this error checking.